impress-org/givewp
# Enhancement: Consolidate test gateways

## Description

There are currently three test gateways in 3.0, which is super confusing. Hahah! There were:
- Test Donations (v2 only)
- Test Gateway
- Test Gateway (offsite)

I tested and realized that Test Gateway is a drop-in replacement for the existing Test Donations gateway, so that can be removed entirely. (Which means less code in `includes`, yay!)

The offsite Test Gateway is really only needed for devs specifically testing how offsite donations work, and is more likely to confuse customers than help them. I think we can add a setting or something in the future if testing offsite gateways comes up more, but for now I simply commented it out from being loaded with a note.

Finally, for the sake of reporting and backwards-compatibility, I changed the Test Gateway to Test Donations and changed its ID to `manual`, which is what the old Test Donations gateway id was. That way, when someone updates to 3.0 there are no surprises or migrations needed, and previous donations made with that gateway will still fit within the new gateway as if nothing had changed.

![there can be only one](https://media.giphy.com/media/8L0yOaWLNmHnm9T4yy/giphy.gif)

## Affects

The aforementioned gateways.

## Visuals

<img width="1046" alt="image" src="https://github.com/impress-org/givewp/assets/2024145/df53c54c-89f8-4e33-b8cd-65e3fd66263a">


## Testing Instructions

Make sure that the Test Donations gateway works with v2 **and** v3 forms.

Also, make some test donations before updating to this, and then update, make some more donations, and make sure the reporting sees them as exactly the same.

## Pre-review Checklist






---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205444026014417
